test/parallel/test-zlib-sync-no-event.js

```diff
@@ -0,0 +1,9 @@
+'use strict';
+require('../common');
+const zlib = require('zlib');
+
+const zipper = new zlib.Gzip();
+zipper.on('close', () => { throw new Error('unexpected `close` event'); });
+
+const buffer = new Buffer('hello world');
+zipper._processChunk(buffer, zlib.Z_FINISH);
```

Comment:
For sake of paranoia, is there an event that would notify that the chunk has been processed and called before `'close'` would be called? I'd like to assert that that event is called in process exit.